File: android/app/src/main/java/net/wellvo/android/ui/components/OnboardingCarousel.kt
package net.wellvo.android.ui.components

import androidx.compose.animation.core.animateDpAsState
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.pager.HorizontalPager
import androidx.compose.foundation.pager.rememberPagerState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import kotlinx.coroutines.launch
import net.wellvo.android.ui.theme.WellvoSpacing

data class OnboardingPage(
    val icon: ImageVector,
    val title: String,
    val body: String,
)

/**
 * A swipeable onboarding carousel with animated dots and a primary CTA.
 * Drop-in replacement for the existing single-screen onboarding intro —
 * host screens can pass their own page list and a completion callback.
 */
@Composable
fun OnboardingCarousel(
    pages: List<OnboardingPage>,
    onComplete: () -> Unit,
    modifier: Modifier = Modifier,
    onSkip: (() -> Unit)? = null,
    primaryCtaLabel: String = "Get Started",
    nextLabel: String = "Next",
    skipLabel: String = "Skip",
) {
    val pagerState = rememberPagerState(pageCount = { pages.size })
    val scope = rememberCoroutineScope()

    Column(
        modifier = modifier
            .fillMaxSize()
            .background(MaterialTheme.colorScheme.background)
            .padding(WellvoSpacing.lg),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        // Top bar with skip
        Box(modifier = Modifier.fillMaxWidth()) {
            if (onSkip != null && pagerState.currentPage < pages.size - 1) {
                TextButton(
                    onClick = onSkip,
                    modifier = Modifier.align(Alignment.CenterEnd),
                ) {
                    Text(skipLabel)
                }
            }
        }

        HorizontalPager(
            state = pagerState,
            modifier = Modifier
                .weight(1f)
                .fillMaxWidth(),
        ) { page ->
            OnboardingPageView(pages[page])
        }

        // Dots
        Row(
            modifier = Modifier.padding(vertical = WellvoSpacing.md),
            horizontalArrangement = Arrangement.spacedBy(8.dp),
        ) {
            repeat(pages.size) { i ->
                val width by animateDpAsState(
                    targetValue = if (i == pagerState.currentPage) 24.dp else 8.dp,
                    label = "dot-width-$i"
                )
                Box(
                    modifier = Modifier
                        .height(8.dp)
                        .width(width)
                        .clip(RoundedCornerShape(4.dp))
                        .background(
                            if (i == pagerState.currentPage)
                                MaterialTheme.colorScheme.primary
                            else
                                MaterialTheme.colorScheme.surfaceVariant
                        )
                )
            }
        }

        Button(
            onClick = {
                if (pagerState.currentPage == pages.size - 1) {
                    onComplete()
                } else {
                    scope.launch { pagerState.animateScrollToPage(pagerState.currentPage + 1) }
                }
            },
            colors = ButtonDefaults.buttonColors(containerColor = MaterialTheme.colorScheme.primary),
            modifier = Modifier
                .fillMaxWidth()
                .height(52.dp),
            shape = RoundedCornerShape(26.dp),
        ) {
            Text(
                text = if (pagerState.currentPage == pages.size - 1) primaryCtaLabel else nextLabel,
                fontWeight = FontWeight.SemiBold,
                style = MaterialTheme.typography.titleMedium,
            )
        }
    }
}

@Composable
private fun OnboardingPageView(page: OnboardingPage) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(horizontal = WellvoSpacing.xl),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center,
    ) {
        Box(
            modifier = Modifier
                .size(160.dp)
                .clip(CircleShape)
                .background(MaterialTheme.colorScheme.primaryContainer),
            contentAlignment = Alignment.Center,
        ) {
            Icon(
                imageVector = page.icon,
                contentDescription = null,
                tint = MaterialTheme.colorScheme.onPrimaryContainer,
                modifier = Modifier.size(80.dp),
            )
        }
        Spacer(Modifier.height(WellvoSpacing.xl))
        Text(
            text = page.title,
            style = MaterialTheme.typography.headlineMedium,
            fontWeight = FontWeight.Bold,
            textAlign = TextAlign.Center,
            color = MaterialTheme.colorScheme.onBackground,
        )
        Spacer(Modifier.height(WellvoSpacing.sm))
        Text(
            text = page.body,
            style = MaterialTheme.typography.bodyLarge,
            textAlign = TextAlign.Center,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
        )
    }
}

File: ios/DailyOK/Views/Components/OnboardingCarousel.swift
import SwiftUI

struct OnboardingPageData: Identifiable {
    let id = UUID()
    let systemImage: String
    let title: String
    let body: String
}

/// A swipeable onboarding carousel with animated dots and a primary CTA.
/// Host screens provide their own page list and a completion callback so
/// the component stays content-agnostic and reusable across the app.
struct OnboardingCarousel: View {
    let pages: [OnboardingPageData]
    var onComplete: () -> Void
    var onSkip: (() -> Void)? = nil
    var primaryCtaLabel: String = "Get Started"
    var nextLabel: String = "Next"
    var skipLabel: String = "Skip"

    @State private var currentIndex: Int = 0

    var body: some View {
        VStack(spacing: 0) {
            HStack {
                Spacer()
                if let onSkip, currentIndex < pages.count - 1 {
                    Button(skipLabel, action: onSkip)
                        .foregroundStyle(DailyOKColor.brand)
                }
            }
            .padding(.horizontal, 20)
            .padding(.top, 16)
            .frame(height: 44)

            TabView(selection: $currentIndex) {
                ForEach(Array(pages.enumerated()), id: \.offset) { index, page in
                    OnboardingPageView(page: page)
                        .tag(index)
                }
            }
            .tabViewStyle(.page(indexDisplayMode: .never))
            .animation(DailyOKMotion.smoothSpring, value: currentIndex)

            HStack(spacing: 8) {
                ForEach(0..<pages.count, id: \.self) { i in
                    Capsule()
                        .fill(i == currentIndex ? DailyOKColor.brand : Color(.systemGray4))
                        .frame(width: i == currentIndex ? 24 : 8, height: 8)
                        .animation(DailyOKMotion.smoothSpring, value: currentIndex)
                }
            }
            .padding(.vertical, 16)

            Button {
                if currentIndex == pages.count - 1 {
                    onComplete()
                } else {
                    withAnimation(DailyOKMotion.smoothSpring) {
                        currentIndex += 1
                    }
                }
            } label: {
                Text(currentIndex == pages.count - 1 ? primaryCtaLabel : nextLabel)
                    .font(.headline.weight(.semibold))
                    .foregroundStyle(.white)
                    .frame(maxWidth: .infinity)
                    .frame(height: 52)
                    .background(
                        Capsule().fill(DailyOKColor.brand)
                    )
            }
            .padding(.horizontal, 24)
            .padding(.bottom, 32)
        }
        .background(Color(.systemBackground))
    }
}

private struct OnboardingPageView: View {
    let page: OnboardingPageData

    var body: some View {
        VStack(spacing: 24) {
            ZStack {
                Circle()
                    .fill(DailyOKColor.green100)
                    .frame(width: 160, height: 160)
                Image(systemName: page.systemImage)
                    .font(.system(size: 72, weight: .semibold))
                    .foregroundStyle(DailyOKColor.green700)
            }
            .accessibilityHidden(true)

            Text(page.title)
                .font(.largeTitle.weight(.bold))
                .multilineTextAlignment(.center)

            Text(page.body)
                .font(.body)
                .multilineTextAlignment(.center)
                .foregroundStyle(.secondary)
        }
        .padding(.horizontal, 32)
    }
}
